src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -22,81 +18,84 @@ class ResourceService(val accessPolicyDAO: AccessPolicyDAO, val directoryDAO: Di
   }
 
   def listUserResourceActions(resourceTypeName: ResourceTypeName, resourceName: ResourceName, userInfo: UserInfo): Future[Set[ResourceAction]] = {
-    listResourceAccessPoliciesForUser(resourceTypeName, resourceName, userInfo).map { matchingPolicies =>
+    listResourceAccessPoliciesForUser(Resource(resourceTypeName, resourceName), userInfo).map { matchingPolicies =>
       matchingPolicies.flatMap(_.actions)
     }
   }
 
   def listUserResourceRoles(resourceTypeName: ResourceTypeName, resourceName: ResourceName, userInfo: UserInfo): Future[Set[ResourceRoleName]] = {
-    listResourceAccessPoliciesForUser(resourceTypeName, resourceName, userInfo).map { matchingPolicies =>
-      matchingPolicies.flatMap(_.role)
+    listResourceAccessPoliciesForUser(Resource(resourceTypeName, resourceName), userInfo).map { matchingPolicies =>
+      matchingPolicies.flatMap(_.roles)
     }
   }
 
-  private def listResourceAccessPoliciesForUser(resourceTypeName: ResourceTypeName, resourceName: ResourceName, userInfo: UserInfo): Future[Set[AccessPolicy]] = {
-    for {
-      policies <- accessPolicyDAO.listAccessPolicies(resourceTypeName, resourceName)
-      groups <- directoryDAO.listUsersGroups(userInfo.userId)
-    } yield {
-      policies.filter { policy =>
-        policy.subject match {
-          case user: WorkbenchUserId => userInfo.userId == user
-          case group: WorkbenchGroupName => groups.contains(group)
-        }
-      }.toSet
+  private def listResourceAccessPoliciesForUser(resource: Resource, userInfo: UserInfo): Future[Set[AccessPolicy]] = {
+    accessPolicyDAO.listAccessPoliciesForUser(resource, userInfo.userId)
+  }
+
+  def listResourcePolicies(resource: Resource): Future[Set[AccessPolicyResponseEntry]] = {
+    accessPolicyDAO.listAccessPolicies(resource).map { policies =>
+      policies.map { policy =>
+        AccessPolicyResponseEntry(policy.name, AccessPolicyMembership(policy.members.members.map(_.value), policy.actions, policy.roles)) //todo
```

Comment:
TODO: this returns the subject ids and group names. we want it to return the emails